### PR TITLE
reorganise gsoc22

### PR DESCRIPTION
**Pull request by Lovkush-A**
the changes include:

- adding a list of contents at the top
- combining 'how to apply' and 'selection process' into one section
- combining 'what we expect from you' and 'what we are looking for' into one section
- adding a timeline to the application process
- convert some paragraphs into bullet lists
- various minor tweaks to the wording

### CODE REVIEW

**Comment by Lovkush-A** [comment]
@fkiraly. I dont seem to be able to add reviewers. Can you have a quick scan to check for any glaring errors?

**Comment by fkiraly** [comment]
Yes - I´ve now added the core developer team as maintainers.

File: internships/gsoc2022.md [hunk 1]
```diff
@@ -1,80 +1,115 @@
...
+## Why sktime?
+Time series data is ubiquitous in many applications. Examples include sensor readings from industrial processes, spectroscopy wave length data from chemical samples, or bed-side monitor medical data from patients. Developing advanced time series analysis capabilities for researchers and practitioners is one of the major challenges of contemporary machine learning.
+
+sktime is a new Python toolbox for machine learning with time series and, to the best of our knowledge, the first unified toolbox for time series. Our ambition is to provide for time series what scikit-learn provides for tabular data. This involves extending scikit-learn to the different time series learning tasks, such as time series classification, clustering, forecasting and anomaly detection. To find out more, check out our [paper](http://learningsys.org/neurips19/assets/papers/sktime_ml_systems_neurips2019.pdf) published at the [Workshop on Systems for ML at NeurIPS 2019](http://learningsys.org/neurips19/).
+
+## The application process
+
+There are two main phases: the sktime application and GSoC proposal, and an interview. Details of these are below. The timeline is:
```
Comment (fkiraly) [inline internships/gsoc2022.md:26]: mentions "two main phases", but then lists three items. Confusing.
Timeline afterwards does not map clearly onto the phases.

File: internships/gsoc2022.md [hunk 2]
```diff
@@ -1,80 +1,115 @@
...
+
+- April 19, 18:00 (UTC). You must complete sktime application form and GSoC proposal by this deadline.
```
Comment (fkiraly) [inline internships/gsoc2022.md:28]: missing: entrance task (PR), condition for the sktime application. Easily confusing.

File: internships/gsoc2022.md [hunk 3]
```diff
@@ -1,80 +1,115 @@
...
+- April 26. We will inform you of the outcome of the first phase *latest* by April 26, but hopefully sooner.
+- April 25 to May 8. Interviews will happen during this time. Preferred dates are April 29 and May 6. Please keep these dates free if possible.
+- May 15. We will take at most one week to inform you of the outcome of the interview. We submit our list of ranked candidates to GSoC committee.
+- May 20. GSoC will inform you of final outcome of the process.
+
+If you get stuck or have questions, please feel free to reach out to us on [Slack](https://join.slack.com/t/sktime-group/shared_invite/zt-62i7aejn-vXc3nOWF26S_P3VXFPWisQ) or [GitHub discussions](https://github.com/alan-turing-institute/sktime/discussions).
+
+### sktime application and GSoC proposal
+
+There are three steps for this. The deadline to complete all of them is April 19, 18:00 (UTC).
```
Comment (fkiraly) [inline internships/gsoc2022.md:38]: confusing - three steps vs two phases.

File: internships/gsoc2022.md [hunk 4]
```diff
@@ -1,80 +1,115 @@
...
+After the deadline on April 19th has passed, we will process the information provided and tell you the outcome of this phase no later than April 26th. There are two possible outcomes:
+- Progress to interview. See below.
+- Rejection. Though this will be disheartening for you, this is only a rejection for GSoC (which is particularly demanding of participants) and should not discourage you from pursuing open source contributions, either with sktime or another package. E.g. you may still be eligible for our (unpaid) 1-1 mentoring.
+
+### Interview
+If successful in the first phase, we will invite you to a structured interview (ca. 30 to 60 minutes length; to be finalised) with core community members of sktime.
+The interview will happen in the two-week period April 26 to May 8, and our preferred dates are April 29 and May 6. Please keep these dates free if possible.
+
+During the interview:
+- we will ask you to give a short (5 to 10 minutes; to be finalised) presentation on a piece of Python code that you wrote
```
Comment (fkiraly) [inline internships/gsoc2022.md:61]: suggest, add the recommendation to come prepared for screen sharing, or with a link to a repo

File: internships/gsoc2022.md [hunk 5]
```diff
@@ -1,80 +1,115 @@
...
+- we will ask about your motivation to join sktime, your previous experience, your suitability for the specified project, and you general technical background in data science and Python.
+
```
Comment (fkiraly) [inline internships/gsoc2022.md:63]: there will also be technical questions

File: internships/gsoc2022.md [hunk 6]
```diff
@@ -1,80 +1,115 @@
...
+* You maintain daily contact with your mentor(s).
+* You engage with the sktime community and other GSoC participants.
+* You peer-review a fellow student's work in the middle and at the end of GSoC.
+* You write weekly blog posts and a final summary post at the end of the project.
+* You consider becoming a *long-term* developer and contributor, stepping up to become a leader in open source toolbox development.
+
+## What you can expect from us
+The expectations on you are high, but you can expect just as much from us. You should expect:
+* We follow the [Code of Conduct](https://www.sktime.org/en/stable/get_involved/code_of_conduct.html).
+* 1-1 mentoring with an experienced contributor.
```
Comment (fkiraly) [inline internships/gsoc2022.md:95]: perhaps add, weekly individual mentoring meeting, max two working day latency in reaction from core devs, daily standups with the community

File: internships/gsoc2022.md [hunk 7]
```diff
@@ -1,80 +1,115 @@
...
+* The opportunity to become a core developer of sktime.
 
-then GSoC with sktime is for you! You'll spend the summer working with our enthusiastic and open-minded team of developers who are creating one of the first comprehensive time series ML toolboxes out there.
 
 ## Projects
 For potential projects, see our [list of suggested projects](https://github.com/sktime/mentoring/blob/main/internships/projects_2022.md) and [good first issues](https://github.com/alan-turing-institute/sktime/labels/good%20first%20issue).
 We also appreciate applications with your own ideas!
 
-## Mentors 2022
+## Mentors
```
Comment (fkiraly) [inline internships/gsoc2022.md:107]: 2022, leave that in - the list changes every year.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks, great work, definitely an improvement.

My main blocking comment is on the "two phases" vs "three steps" and obscurity around the requirements to complete the application. Added  details on that, and additional other comments, above.

File: internships/gsoc2022.md (hunk 6 above)
Comment (Lovkush-A) [reply internships/gsoc2022.md:95]: done

File: internships/gsoc2022.md (hunk 5 above)
Comment (Lovkush-A) [reply internships/gsoc2022.md:63]: clarified final phrase to make this clear

File: internships/gsoc2022.md (hunk 2 above)
Comment (Lovkush-A) [reply internships/gsoc2022.md:28]: adding entrance task in the relevant places

**Comment by Lovkush-A** [comment]
@fkiraly I have made changes according to all the things you suggested. Ready for a second review!

**Review by fkiraly — APPROVED** [review]
looks good, thanks!

**Pull request merged**